### PR TITLE
Add support for ephemeral containers

### LINKED ISSUE #1103: Create ephemeral containers for debugging
**Describe the problem/challenge you have**
Some containers might not have bash or sh for a container exec.

**Describe the solution you'd like**
An implementation of 
https://kubernetes.io/docs/tasks/debug-application-cluster/debug-running-pod/#ephemeral-container

This feature could have the following additions:
 - Failing to create a terminal when navigating to the tab prompts user to create an ephemeral container.
 - Modify object status to remind users to delete pods with a running ephemeral container.

**Anything else you would like to add:**
n/a

**Environment:**

- Octant version (use `octant version`): 0.13.1
- Kubernetes version (use `kubectl version`): 1.18

### PR DESCRIPTION
**Pull request by GuessWhoSamFoo**
This PR contains an initial pass at supporting ephemeral containers in Octant. If the feature gate is enabled when visiting a pod, it will automatically add debug container using a debian image.

Pods with an ephemeral container will have their object status modified as `warning` (yellow).

If there is interest in this feature, further efforts can be invested into making the debug container configurable and proper UI.

**Which issue(s) this PR fixes**
- Fixes #1103 

**Special notes for your reviewer**:
Example of a distroless image where `sh` and `bash` will not work:

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx-deployment
spec:
  selector:
    matchLabels:
      app: nginx
  replicas: 3
  template:
    metadata:
      labels:
        app: nginx
    spec:
      shareProcessNamespace: true
      containers:
      - name: nginx
        image: sfoo/nginx-distroless:latest
        ports:
        - containerPort: 80
        - containerPort: 443
```

Testing if a feature gate is enabled using mocked clients is not straightforward and I haven't figured out a way to do this yet. The terminal viewer component is pending a refactor soon because the terminal viewer does not have knowledge of the manager's state which causes unnecessary synchronization challenges.

Signed-off-by: GuessWhoSamFoo <foos@vmware.com>

### CODE REVIEW

File: internal/api/terminal_manager.go [hunk 1]
```diff
@@ -136,16 +136,24 @@ func (s *terminalStateManager) startStream(key store.Key, container string) cont
...
 	eventType := event.NewTerminalEventType(key.Namespace, key.Name, container)
 
-	instance, err := terminal.NewTerminalInstance(ctx, s.config.ClusterClient(), logger, key, container, "/bin/sh", s.chanInstance)
-	if err != nil {
-		cancelFn()
-		return cancelFn
-	}
-
-	s.instance = instance
+	commands := []string{"sh", "bash", "powershell", "cmd"}
```
Comment (wwitzel3) [inline internal/api/terminal_manager.go:139]: This should be it's own PR with a changelog entry.

**Review by wwitzel3 — CHANGES_REQUESTED** [review]
This looks pretty good, part of this should be a PR that lands in upstream as its own change though.

File: internal/api/terminal_manager.go (hunk 1 above)
Comment (GuessWhoSamFoo) [reply internal/api/terminal_manager.go]: Removed and tracked in #1749

**Review by wwitzel3 — APPROVED** [review]
